src/TeleportJoin.sol

```diff
@@ -211,6 +237,20 @@ contract TeleportJoin {
         emit Mint(hashGUID, teleportGUID, amtToTake, maxFeePercentage, operatorFee, msg.sender);
     }
 
+    /**
+    * @dev External authed function that registers the teleport
+    * @param teleportGUID Struct which contains the whole teleport data
+    **/
+    function registerMint(
+        TeleportGUID calldata teleportGUID
+    ) external auth {
```

Comment:
Instead of `auth` it would be more explicit to just `require(msg.sender == router)`

Reply:
Router may not be the only source. In the same way `requestMint()` is called from the Oracle Auth maybe we want to do the same for `registerMint()` in the future.

Reply:
Yeah I vote to always keep `auth` excepting special cases like in the D3M we had to do the `isHub` for after ES situation.